### PR TITLE
adding options to pod execution

### PR DESCRIPTION
**Pull request by mshitrit**
adding options to pod execution:
- allowing execution of commands without expected output
- allowing using a user defined pod executing

### CODE REVIEW

File: test/command/command.go [hunk 1]
```diff
@@ -20,11 +20,18 @@ import (
...
 
+// RunOptions allows additional options
+type RunOptions struct {
+	IsOutputExpected bool
+	ExecutePod       *corev1.Pod
+}
+
 // RunCommandInCluster runs a command in a pod in the cluster and returns the output
-func RunCommandInCluster(ctx context.Context, c *kubernetes.Clientset, nodeName string, ns string, command string, log logr.Logger) (string, error) {
+func RunCommandInCluster(ctx context.Context, c *kubernetes.Clientset, nodeName string, ns string, command string, log logr.Logger, opts ...*RunOptions) (string, error) {
```
Comment (clobrano) [inline test/command/command.go]: Please update the function description, since now the output is not always returned.

**Review by clobrano — CHANGES_REQUESTED** [review]
I think the function description should be updated to clarify that only the 1st RunOptions is actually used (as the variadic argument is used to allow no RunOptions). Otherwise, you might think of defining two APIs, one with 1 RunOptions (user defined) and one without RunOptions, which internally calls the first passing the default RunOptions.

File: test/command/command.go (hunk 1 above)
Comment (mshitrit) [reply test/command/command.go]: Hmm interesting question.
I think the output is still always return, the only difference is that empty output can be considered valid via optional RunOption.

WDYT ?

Comment (clobrano) [reply test/command/command.go]: Good point, however it should be explained in the docstring, together with the fact that the function doesn't actually takes multiple `RunOptions` (IIUC)

Comment (mshitrit) [reply test/command/command.go]: > together with the fact that the function doesn't actually takes multiple RunOptions

I think this is an excellent point, I've modified the code so it'll take multiple RunOption, which I think is the standard pattern.

>it should be explained in the docstring

done

**Review by clobrano — APPROVED** [review]

**Comment by clobrano** [comment]
/lgtm
Giving a chance to get more reviews, feel free to unhold
/hold

**Pull request merged**